src/devices/stepper.rs

```diff
@@ -0,0 +1,526 @@
+            ffi::PhidgetStepper_getMinCurrentLimit(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+    /// Get maximum current limit
+    pub fn get_max_current_limit(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMaxCurrentLimit(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Set data interval
+    pub fn set_data_interval(&self, data_interval: u32) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setDataInterval(self.chan, data_interval)
+        })?;
+        Ok(())
+    }
+    /// Get data interval
+    pub fn get_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getDataInterval(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum data interval
+    pub fn get_min_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMinDataInterval(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Get maximum data interval
+    pub fn get_max_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMaxDataInterval(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Set data rate
+    pub fn set_data_rate(&self, data_rate: f64) -> Result<()> {
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_setDataRate(self.chan, data_rate) })?;
+        Ok(())
+    }
+    /// Get data rate
+    pub fn get_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum data rate
+    pub fn get_min_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMinDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get maximum data rate
+    pub fn get_max_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMaxDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Set engaged
+    pub fn set_engaged(&self, engaged: bool) -> Result<()> {
+        let _v = match engaged {
+            true => 1,
+            false => 0,
+        };
+
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_setEngaged(self.chan, _v) })?;
+        Ok(())
+    }
+
+    /// Get engaged
+    pub fn get_engaged(&self) -> Result<bool> {
+        let mut _v = 0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getEngaged(self.chan, &mut _v) })?;
+
+        match _v {
+            1 => Ok(true),
+            0 => Ok(false),
+            _ => Err(ReturnCode::UnknownVal),
+        }
+    }
```

Comment:
In C/C++, any non-zero value of an intger type is considered `true`. Most commonly it is 1, but it could be anything. This is especially true in hardware code where the boolean value might a bit in a register at some arbitrary position in the word which the function just masks. 

For example, here, the "engaged" flag might be bit 2 in the motor's status register. So the C library might return this as:
```
  return status_reg & (1 << 2);
```
So, a `true` would be 0x04 and `false` would be zero. (I'm making this up as an example, but this is a common C implementation practice).

So, to convert an integer from a C library to a bool, you should use the C convention for zero is false, and anything else is true. Convert it here as `v != 0`, like:

```
pub fn get_engaged(&self) -> Result<bool> {
    let mut v = 0;
    ReturnCode::result(unsafe { ffi::PhidgetStepper_getEngaged(self.chan, &mut v) })?;
    Ok(v != 0)
}

Reply:
Good to know